x/gamm/keeper/grpc_query.go

```diff
@@ -184,14 +175,14 @@ func (q Querier) PoolsWithFilter(ctx context.Context, req *types.QueryPoolsWithF
 		checks_needed++
 	}
 
-	for _, p := range pools {
+	var response = []*codectypes.Any{}
+	pageRes, err := query.FilteredPaginate(poolStore, req.Pagination, func(_, value []byte, accumulate bool) (bool, error) {
```

Comment:
Using `FilteredPaginate` instead of `Paginate` method allows us to iterate until pagination count has been reached

Reply:
https://github.com/cosmos/cosmos-sdk/issues/12242

We can backport the fix for that to clean this up a little if we want.
(Or just copy/paste https://github.com/cosmos/cosmos-sdk/pull/12253 into osmoutils)

Reply:
I was also checking generic pagination as well! Sad that we don't have it yet. I say we refactor this to use generic pagination later on when it gets backported and for now leave as is